nixos/modules/programs/goldwarden.nix

```
@@ -0,0 +1,36 @@
+{ lib, config, pkgs, ... }:
+let
+  cfg = config.programs.goldwarden;
+in
+{
+  options.programs.goldwarden = {
+    enable = lib.mkEnableOption "Goldwarden";
+    package = lib.mkPackageOption pkgs "goldwarden" {};
+  };
+
+  config = lib.mkIf cfg.enable {
```

```suggestion
  config = lib.mkIf cfg.enable {
      assertions = [{ 
         assertion = !config.programs.ssh.startAgent;
         message = "Only one ssh-agent can be used at a time.";
      }];
```

Optionally also include another Option for not setting `SSH_AUTH_SOCK`, incase a User might want to use a different agent.